weather_dl/download_pipeline/clients.py

```diff
@@ -54,7 +56,7 @@ def __init__(self, config: Config, level: int = logging.INFO) -> None:
         self.logger.setLevel(level)
 
     @abc.abstractmethod
-    def retrieve(self, dataset: str, selection: t.Dict, output: str) -> None:
+    def retrieve(self, dataset: str, selection: t.Dict, output: str, manifest: Manifest) -> None:
```

Comment:
By default, it will use NoOpManifest since this argument is passed from fetcher.py, and if we check there, we can see that the manifest is initialized to NoOpManifest by default."

Reply:
SGTM, thanks!